Work on adding show options to code list pop up

Crabcyborg opened the pull request:
Related update https://github.com/Strategy11/formidable-pro/pull/4573 (merged)

Newer related update https://github.com/Strategy11/formidable-pro/pull/6549

**TODO**
- [x] Add show=label options for fields with options as well.

<img width="452" alt="image" src="https://github.com/user-attachments/assets/35ac8819-d590-4b38-95ee-6dd5956a1a9f">

### Test procedure
1. Create a form and add a Text field, Name field and an option field like Dropdown field.
2. For the Dropdown field, change one of the labels to a number, like 10 for example.
3. For the option fields select 'Use separate values' option.
4. Go to the Text field options and under Advanced > Default Value click on the Calculation tab.
5. Click on the three dots to trigger the field shortcodes popup and Select 'Text' tab for the calculation.
6. Confirm that the name field now has three more shortcode options that allows inserting a name part. Ex. `show=first`
7. Insert one of the new show shortcodes in the calculation.
8. Save and Preview the form.
9. Confirm that everything works as expected.
10. Turn back to the form builder and check that an additional `show=label` option is available for the Dropdown field.
11. Now, change the calculation of the text field to from 'Text' to 'Math' using the tab and insert the show=label shortcode into the calculation and build a number calculation from that adding more operands.
12. Save and Preview the form.
13. Confirm that the number calculation works as expected when a number label is selected for the Dropdown field.

<img width="473" alt="image" src="https://github.com/user-attachments/assets/60dd310d-9567-4bdf-876d-1f87180249f4">


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added shortcode options for individual parts of multi-part fields, including configured first, middle, and last name components.
  * Added support for extensions to provide shortcode parts for other field types.
  * Added field type, label, and styling details to generated shortcode options.

* **Bug Fixes**
  * Preserved custom option labels without truncation.
  * Corrected the display order and localization of name-part labels.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Crabcyborg committed:
Work on adding show options to code list pop up

Crabcyborg commented:
@AbdiTolesa I was working on adding support for `show=first`/`show=last` to the shortcode pop up for name fields to make it easier to insert these special options.

Do you think you'd be able to try taking this over and wrapping it up? I was hoping to also include `show="label"` for the fields with options that have "Use separate values" enabled as well.

Thank you!

AbdiTolesa committed:
Fix merge conflict and merge master

AbdiTolesa committed:
Add show=label shortcode types to calculation fields popup

AbdiTolesa committed:
Fix eslint error

AbdiTolesa committed:
Move name field additional shortcode variants implementation to pro builder

AbdiTolesa committed:
Undo last commit

AbdiTolesa committed:
Refactor code a bit

AbdiTolesa committed:
Add a return type definition

AbdiTolesa committed:
Reduce nesting complication

AbdiTolesa committed:
Add condition check for showing show=xx shortcodes in calculation popup

AbdiTolesa committed:
Add inline comment to clarify condition check

AbdiTolesa committed:
Add Rename function

AbdiTolesa committed:
Fix eslint error

truongwp commented on js/formidable_admin.js:
```diff
@@ -2867,24 +2867,94 @@ function frmAdminBuildJS() {
-			li.appendChild( a );
-			list.appendChild( li );
+			addCalcFieldLiToList( list, fieldId, fields[i].fieldId, fields[i].fieldName, fields[i].fieldType );
+			if ( document.getElementById( 'new_fields' ).dataset.supportsShowShortcodes ) { // If the required Pro updates are there to support "show" shortcodes.
```
`document.getElementById( 'new_fields' ).dataset.supportsShowShortcodes` should be moved to the outside of the loop.

truongwp commented on js/formidable_admin.js:
```diff
@@ -2867,24 +2867,94 @@ function frmAdminBuildJS() {
+				document.createTextNode( label )
+			]
+		});
+		anchor.setAttribute( 'data-code', code );
```
We can set `data-code` in `a()` function with the `data`.

AbdiTolesa committed:
Improve performance

truongwp reviewed (approved):
This looks good. Thanks @AbdiTolesa!

AbdiTolesa commented:
@Crabcyborg This is approved by Garret and Truong but I couldn't request your review since you own the PR originally. Could you review it please?

Crabcyborg commented on js/formidable_admin.js:
```diff
@@ -2854,35 +2854,109 @@ function frmAdminBuildJS() {
+	 * @param {string} fieldType
+	 * @param {Number} fieldId
+	 *
+	 * @returns {Boolean}
```
@AbdiTolesa

If `document.getElementById( `separate_value_${fieldId}` )` is undefined then this would actually return `undefined` and not a boolean.

AbdiTolesa replied:
You're right, we need to get the boolean value of that and I have done that now.

Crabcyborg committed:
Merge branch 'master' into work_on_adding_show_options_to_code_list_popup

AbdiTolesa committed:
Make sure a function always return boolean

AbdiTolesa committed:
Merge branch 'work_on_adding_show_options_to_code_list_popup' of https://github.com/Strategy11/formidable-forms into work_on_adding_show_options_to_code_list_popup

Crabcyborg committed:
Merge branch 'master' into work_on_adding_show_options_to_code_list_popup

Crabcyborg commented:
@AbdiTolesa Sorry I let this PR get so old.

We've updated the admin JS now so it's in a different admin.js file, and this file that's being changed is minified now.

So we'll need to move these changes into the new file.

Could you help with that?

Thank you!

AbdiTolesa committed:
Merge master

AbdiTolesa committed:
Move changes to source file

AbdiTolesa committed:
Build files

AbdiTolesa committed:
Fix eslint errors

AbdiTolesa commented:
> So we'll need to move these changes into the new file.
> 
> Could you help with that?

@Crabcyborg I have moved the changes to the source js files both in this PR and https://github.com/Strategy11/formidable-pro/pull/5394

AbdiTolesa committed:
Get rid of unnecessary checks dependent on Pro

shervElmi committed:
Add Address field part shortcodes to calculation popup

- Added maybeAddAddressPartShortcodes function to support Address field part shortcodes (line1, line2, city, state, zip, country)
- Added function call in field population loop to display Address field parts in calculation popup
- Similar pattern to Name field implementation

Resolves #5825

shervElmi committed:
Refactor: Improve maybeAddAddressPartShortcodes with modern ES6 syntax

- Use template literals instead of string concatenation
- Extract addressParts as const variable for better readability
- Add descriptive JSDoc parameter documentation
- Refine function description to be more professional and direct
- Cleaner forEach syntax with consistent formatting

shervElmi committed:
Refactor: Clean up JSDoc parameter formatting in maybeAddAddressPartShortcodes

- Remove hyphens from JSDoc parameter descriptions for consistency with project style
- Align parameter documentation format with standard JSDoc conventions

shervElmi committed:
Add extensibility hook for field part shortcodes in calculation popup

- Added wp.hooks.doAction('frm_add_calc_field_shortcodes') hook
- Removed maybeAddAddressPartShortcodes() function (Address fields are Pro-only)
- Allows Pro and other extensions to add custom field part shortcodes
- Well-documented hook with JSDoc for extension developers

Crabcyborg committed:
Merge pull request #2639 from Strategy11/feature/address-field-show-options

Add Address field part shortcodes to calculation popup

AbdiTolesa committed:
Rebuild files

AbdiTolesa committed:
Rebuild files

AbdiTolesa committed:
Rebuild files

AbdiTolesa committed:
Rebuild files and merge master

Crabcyborg committed:
Field shortcode popup improvements

Crabcyborg committed:
Merge branch 'master' into work_on_adding_show_options_to_code_list_popup

Crabcyborg committed:
Run build

Crabcyborg committed:
Exclude from math calc options
